draft-ietf-oauth-transaction-tokens.md

```diff
@@ -584,14 +584,23 @@ To request a replacement Txn-Token, the requester makes a Txn-Token Request as d
 A successful response by the Txn-Token Service to a Replacement Txn-Token Request is a Txn-Token Response as described in {{txn-token-response}}
 
 ## Mutual Authentication of the Txn-Token Request
+A workload and Transaction Token Service MUST perform mutual authentication using strong authentication mechanisms.
+
 A Txn-Token Service MUST ensure that it authenticates any workloads requesting Txn-Tokens. In order to do so:
 
-* It MUST name a limited, pre-configured set of workloads that MAY request Txn-Tokens
-* It MUST individually authenticate the requester as being one of the named requesters
-* It SHOULD rely on mechanisms, such as {{SPIFFE}} used in conjunction with MTLS {{RFC8446}}, or some other means of performing MTLS, to securely authenticate the requester
-* It SHOULD NOT rely on insecure mechanisms, such as long-lived shared secrets to authenticate the requesters
+* It MUST maintain a limited, pre-configured set of authorized workloads that MAY request Txn-Tokens.
+* It MUST authenticate the requesting workload and confirm that it is included in the list of workloads authorized to request a transaction token.
+* It SHOULD accept workloads credentials such as JWTs or X.509 certificates which MAY be provisiond using mechanisms such as {{SPIFFE}} or other provisioning protocols.
+* It SHOULD use X.509 or JWT credentials in conjunction with MTLS {{RFC8446}}, or another secure authentication protocol, to securely authenticate the requesting workload.
+* It SHOULD NOT rely on insecure mechanisms, such as long-lived shared secrets to authenticate the requesting workloads.
+
+The requesting workload MUST ensure that it authenticated the Transaction Token Service. In order to do so:
 
-The requesting workload MUST have a pre-configured location for the Transaction Token Service. It SHOULD rely on mechanisms, such as {{SPIFFE}}, to securely authenticate the Transaction Token Service before making a Txn-Token Request.
+* It MUST have a pre-configured location for the Transaction Token Service.
+* If there is more than one Transaction Token Service, it MUST ensure that it is sending transaction token requests to the right Transaction Token Service to avoid leaking sensitive information like access tokens.
```

Comment:
@gffletch I think you pointed this out in one of the issues- are you OK with removing this bullet given the limitation in Section 6?

Reply:
We've talked about a logical TTS which doesn't require a single endpoint. 

Otherwise our text about the requesting workload needing to ensure it is talking to the correct TTS should cover this.

(I am doing this from my phone:)

Reply:
OK, so my take here is that we stick with a single logical TTS. I will add a security consideration to remind people why they should make sure they request transaction tokens to the right server (and not some rogue) and delete line 600.

Reply:
Updated text